cdap-proto/src/main/java/io/cdap/cdap/proto/sourcecontrol/RepositoryConfig.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.proto.sourcecontrol;
+
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * Represents the repository configuration of a namespace. This class needs to be GSON serializable.
+ */
+public class RepositoryConfig {
+  private final Provider provider;
+  private final String link;
+  private final String defaultBranch;
+  private final String pathPrefix;
+  private final AuthConfig auth;
+
+  RepositoryConfig(Provider provider, String link, String defaultBranch, AuthConfig authConfig,
```

Comment:
usually when there is a builder, the constructor is private. That way we can ensure everything goes through the builder. If this is package private because we want to create invalid configs for testing, then add a `@VisibleForTesting` annotation.

Reply:
Generally it seems to imply not a very good design to put `@VisibleForTesting` annotation to test the validation logic. Instead I added `RepositoryConfigTest` to cover the validation paths